### PR TITLE
SQL extensions: add support for DROP IF EXISTS

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala [hunk 1]
```diff
@@ -21,24 +21,33 @@ import org.apache.spark.sql.connector.catalog.CatalogPlugin
...
     branch: String,
     currentCatalog: CatalogPlugin,
     isBranch: Boolean,
-    catalog: Option[String]
+    catalog: Option[String],
+    failOnDrop: Boolean
 ) extends NessieExec(catalog = catalog, currentCatalog = currentCatalog) {
 
   override protected def runInternal(
       api: NessieApiV1
```
Comment (dimas-b) [inline integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala:36]: Just wondering: Can we switch to API v2 here (separate PR, of course)?

File: site/docs/tools/sql.md [hunk 2]
```diff
@@ -65,6 +65,8 @@ Dropping a tag `devTag` in the `nessie` catalog (in case it exists):
 
 * `DROP TAG IF EXISTS devTag IN nessie`
 
+Note: the `IF EXISTS` clause is optional and is only supported for Nessie 0.72 or higher.
```
Comment (dimas-b) [inline site/docs/tools/sql.md:68]: Maybe `Nessie SQL Extentions 0.72 ...`. This change looks purely client-side.

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala (hunk 1 above)
Comment (adutra) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala:36]: Keeping in mind that `NessieAPIV2` extends `NessieAPIV1`, what is the immediate gain of forcing this code to only work with v2? From my (little) experience, Iceberg 1.3.1 is still on v1, right? Wouldn't your suggestion create issues with Iceberg < 1.4?

**Review by dimas-b — APPROVED** [review]

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala (hunk 1 above)
Comment (dimas-b) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala:36]: This should work in Iceberg 1.3 too, AFAIK. API v2 has been around for a long time... but I did not check specifically :)

Comment (dimas-b) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala:36]: I mean `NessieCatalog` in Iceberg 1.3 does use API v1, but v2 client should be available, I guess.

Comment (ajantha-bhat) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala:36]: switching to Nessie V2 client is not possible from Iceberg 1.3.

Possible from Iceberg 1.4 with the catalog property. 
ref: https://github.com/apache/iceberg/pull/6712

Comment (dimas-b) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala:36]: let's defer this until Iceberg 1.4 is released :)

**Review by ajantha-bhat — APPROVED** [review]

**Comment by bigluck** [comment]
Thanks guys!!

**Pull request merged**